spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/Memoizer.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+/**
+ * Memorize function execution result
+ *
+ * @author Warren Zhu
+ */
+public class Memoizer {
```

Comment:
public final class Memoizer?

Reply:
also add a private constructor to this class

Reply:
try `@NoArgsConstructor(access = AccessLevel.PRIVATE)` here.